terraform/environments/preview/s3_buckets.tf

```diff
@@ -0,0 +1,168 @@
+      type        = "AWS"
+    }
+
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:PutObjectAcl",
+      "s3:GetBucketLocation",
+      "s3:ListBucket",
+    ]
+
+    resources = [
+      "arn:aws:s3:::digitalmarketplace-agreements-preview-preview/*",
+      "arn:aws:s3:::digitalmarketplace-agreements-preview-preview",
+    ]
+  }
+}
+
+resource "aws_s3_bucket" "agreements_bucket" {
+  bucket = "digitalmarketplace-agreements-preview-preview"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
+
+  logging {
+    target_bucket = "${aws_s3_bucket.server_access_logs_bucket.id}"
+    target_prefix = "digitalmarketplace-agreements-preview-preview/"
+  }
+
+  policy = "${data.aws_iam_policy_document.agreements_bucket_policy_document.json}"
+}
+
+# Communications jenkins: read write
+
+data "aws_iam_policy_document" "communications_bucket_policy_document" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      identifiers = ["arn:aws:iam::${var.aws_main_account_id}:role/jenkins-ci-IAMRole-1FIPDG9DE2CWJ"]
+      type        = "AWS"
+    }
+
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:PutObjectAcl",
+    ]
+
+    resources = [
+      "arn:aws:s3:::digitalmarketplace-communications-preview-preview/*",
+      "arn:aws:s3:::digitalmarketplace-communications-preview-preview",
+    ]
+  }
+}
+
+resource "aws_s3_bucket" "communications_bucket" {
+  bucket = "digitalmarketplace-communications-preview-preview"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
+
+  logging {
+    target_bucket = "${aws_s3_bucket.server_access_logs_bucket.id}"
+    target_prefix = "digitalmarketplace-communications-preview-preview/"
+  }
+
+  policy = "${data.aws_iam_policy_document.communications_bucket_policy_document.json}"
+}
+
+# Documents - jenkins: read write list
+
+data "aws_iam_policy_document" "documents_bucket_policy_document" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      identifiers = ["arn:aws:iam::${var.aws_main_account_id}:role/jenkins-ci-IAMRole-1FIPDG9DE2CWJ"]
+      type        = "AWS"
+    }
+
+    actions = [
+      "s3:GetObject",
+      "s3:PutObject",
+      "s3:PutObjectAcl",
+      "s3:GetBucketLocation",
+      "s3:ListBucket",
+    ]
+
+    resources = [
+      "arn:aws:s3:::digitalmarketplace-documents-preview-preview/*",
+      "arn:aws:s3:::digitalmarketplace-documents-preview-preview",
+    ]
+  }
+}
+
+resource "aws_s3_bucket" "documents_bucket" {
+  bucket = "digitalmarketplace-documents-preview-preview"
+  acl    = "private"
+
+  versioning {
+    enabled = true
+  }
+
+  logging {
+    target_bucket = "${aws_s3_bucket.server_access_logs_bucket.id}"
+    target_prefix = "digitalmarketplace-documents-preview-preview/"
+  }
+
+  policy = "${data.aws_iam_policy_document.documents_bucket_policy_document.json}"
+}
+
+# G7-draft-documents
+
+resource "aws_s3_bucket" "g7-draft-documents_bucket" {
```

Comment:
No policy for this bucket?

Reply:
![screen shot 2018-07-26 at 15 36 41](https://user-images.githubusercontent.com/3469840/43269005-db322a40-90e9-11e8-8670-5c4306e6ca91.png)

Reply:
![screen shot 2018-07-26 at 15 36 58](https://user-images.githubusercontent.com/3469840/43269041-eb2e23a4-90e9-11e8-8bc0-d8f613b06ebd.png)

Reply:
Nada 🤔  Bit weird